File: CS333 - Programming Languages/Introduction to non-C Languages/C++/hello_world.cpp
/*
 * File: hello_world.cpp
 * Prupose: To print the string 'hello world' and other command line arguments
 * Author: Samuel Munoz
 * Date: 02-24-2020
 */

// #include is a keyword that allows the programmer to import packages in the C/C++ libraries
#include <iostream>

// this the main method that runs when this code is compiled and executed
int main(int argc, char *argv[]) {
	
	// std::cout is used to print data types to the console
	// std::endl is used to show when a line end and print a new line
	std::cout << "Hello World!" << std::endl;
	
	std::cout << "Command Line Argments:" << std::endl;
	
	// this is a for loop. The syntax for this loop is exactly the same for languages like Java
	for(int i=0;i<argc-1;i++) {
		std::cout << i+1 << ": " << argv[i+1] << std::endl;
	}
	
	// this calls the exit status in C++ so C++ knows when the program ends
	return 0;
}

File: CS333 - Programming Languages/Introduction to non-C Languages/C#/HelloWorld.cs
/*
* File: Program.cs
* Purpose: To create a hello world program and print out command line arguments
* Author: Samuel Munoz
* Date: 02/26/2020
*/

// NOTE: this code was in part created by the dotnet framework which explains why some of this code where exist

// the keyword 'using' allows C# to import packages from the C# library
using System;

// the keyword 'namespace' is used to link this Program file is linked to the HelloWorld dotnet project I created
namespace HelloWorld
{
    class Program
    {
        // here is where main method that runs at runtime
        static void Main(string[] args)
        {
            // Console.WriteLine allows C# to print data types onto the console
            Console.WriteLine("Hello World!");
            Console.WriteLine("Command Line Arguments:");
            // for loops and the syntax for 'for' loops is the same as Java code
        	for(int i=0;i < args.Length;i++) {
                // one aspect about Console.WriteLine is that WriteLine can print any type of data, not just strings
        		Console.WriteLine(i + ": " + args[i]);
        	}
        }
    }
}
